client/src/util/fetchFromAPI.test.js

```diff
@@ -32,15 +32,16 @@ it("mocks the fetch request", async () => {
 // });
 
 describe("test should return a failed response", () => {
-  it("testing the fetch condition", async () => {
+  it("testing fetch condition", () => {
     fetch.mockReject(new Error("error message"));
-    const result = { data: null, error: null };
-    return await queryAPI({
-      url: "/",
-      onSuccess: jest.fn(),
-      onError: jest.fn(),
-    }).then(() => {
-      expect(fetch).toEqual(result.error);
-    });
+    const onSuccess = jest.fn();
+    const onError = jest.fn();
+    return queryAPI("/")
+      .then(onSuccess)
+      .catch(onError)
+      .finally(() => {
+        expect(onSuccess).not.toHaveBeenCalled();
+        expect(onError).toHaveBeenCalled();
```

Comment:
Validate that onError was called with the message `error message`.

Reply:
@brentmclark pushed and test has passed ...I'm I testing the correctly or rather what is intended to be tested?

<img width="329" alt="Screen Shot 2023-03-14 at 11 24 43 AM" src="https://user-images.githubusercontent.com/34128735/225051423-ea84c00d-f2ca-4dea-9aa9-3fb5cd673dc7.png">
<img width="347" alt="Screen Shot 2023-03-14 at 11 24 52 AM" src="https://user-images.githubusercontent.com/34128735/225051448-4d846300-4566-489e-8304-dcc7fc140b3b.png">